src/terminal-screen.c

```diff
@@ -1684,33 +1690,57 @@ terminal_screen_button_press (GtkWidget      *widget,
 	TerminalScreen *screen = TERMINAL_SCREEN (widget);
 	gboolean (* button_press_event) (GtkWidget*, GdkEventButton*) =
 	    GTK_WIDGET_CLASS (terminal_screen_parent_class)->button_press_event;
-	char *matched_string;
-	int matched_flavor = 0;
+	char *hyperlink;
+	char *url;
+	int url_flavor = 0;
 	guint state;
 
 	state = event->state & gtk_accelerator_get_default_mod_mask ();
 
-	matched_string = terminal_screen_check_match (screen, (GdkEvent*)event, &matched_flavor);
+	hyperlink = terminal_screen_check_hyperlink (screen, (GdkEvent*)event);
+
+	if (hyperlink != NULL &&
+		(event->button == 1 || event->button == 2) &&
+		(state & GDK_CONTROL_MASK))
+	{
+		gboolean handled = FALSE;
 
-	if (matched_string != NULL &&
+		g_signal_emit (screen, signals[MATCH_CLICKED], 0,
+	                 hyperlink,
+	                 FLAVOR_AS_IS,
```

Comment:
Since you pre-assigned `url_flavor` to `0` (which is `FLAVOR_AS_IS`), maybe send the `url_flavour` here to keep it consistent. You will override it anyway if it's not handled here.

Reply:
as I found later - url_flavor is not assigned for hyperlinks